### DISCUSSION TITLE
Dexcome One sensor not starting on xDrip with Samsung Galaxy S5 mini

### DISCUSSION THREAD

**Discussion by kresimirjedud** [NightscoutFoundation/xDrip#2663] [Q&A]
Hi,
I cannot start a new Dexcome one sensor in xDrip on Samsung Galaxy S5 mini. I am using Dexcome one for the first time ever. The transmitter serial starts with C, but I have the latest xDrip.

Here are the specifications:
**PHONE**
Samsung Galaxy S5 mini
Model SM-G800F
Android 6.0.1

**XDRIP**
b3ba148-2023.02.07

I have followed all the instructions when setting up xDrip, and have double-checked them with NavidFo as well. He pointed me to start a discussion here. I am aware this is an old mobile phone version, but it is the smallest one on the market (since the child is using it). Also, I have been successfully using xDrip on it for a long time, but with Libre1 and Miaomiao. Miaomiao died so we decided to try with Dexcome since people say they are satisfied with it and Miaomiao had a bad battery issue and gave us a hard time .... 

Thank you in advance!

**Comment by jamorham** [comment]
Please enable OB1 debugging as described here: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector and go through the process of trying to get it to work and then submit me the logs using the event viewer logs upload feature.

I suspect that it will be related to the age of the phone and android version which is the likely source of the problem (I haven't tested recently below android 8) but I will wait to see what the logs show.

**Reply by kresimirjedud** [reply]
Hi,
I sent you 2 logs. Please discard the first one and see if the second one can help. I saw that the phone was trying to connect to other bt devices such as Miband and recuperator so I turned them off, went to a separate room where their signal cannot reach, cleared bt cache in phone system and restarted the phone and then initialized connection as Navid explained (by changing and ten returning the transmitter id) but no luck. Just one more thing - I use this phone only because it is small and practical for the child to have it.  If it si too complicated to get it work that's fine, I'll get another phone, but unfortunately all newer phones are large and unpractical for a child. Also, I used xDrip on this phone with miaomiao earlier with no problem so I thought it would function out-of.a-box ...
Thank you once again for your effort!

**Reply by jamorham** [reply]
Your logs seem to show that for an amount of time it is searching for a transmitter id ending in C but yours ends in V so check that is entered correctly.

From the logs it appears that you don't complete pairing. Do you get a pop-up asking to pair that you accept? Check that the phone is not on silent or do-not-disturb as this could hide the pairing request. Respond to the pairing request as quickly as you can.

I've recently been testing a Cubot KingKong Mini2 which is a small form factor phone. So far it seems to be working fine (phone rom version 20220119), but this was sent to me because someone was reporting problems with one, so mileage may vary.. the S5 mini is really very old now so you could consider as an alternative if you can't get that working.

**Reply by kresimirjedud** [reply]
@jamorham thank you for your reply. Regarding the transmiter ending in C this is due to Navid's instructions to initialise pairing - the instruction was to change the ID, save and then change again to the right id. It obviously did not help.
Regarding pairing I don't get a pairing request at all. Perhaps that is the main issue. Phone is not on do-not-disturb, and it was on silent (sound) and I turned the sound on but no help. 

This really is an old phone and I wouldn't want to waste your time in researching the ancient history. I do like a Cubot KingKong Mini2, this is exactly what I need. Regarding the cubot I only have one more question: do you think it will be possible to install AAPS on it as well? If so I'm closing this discussion and going directly to the shop ... Thank you for your time and effort!

**Reply by Navid200** [reply]
I don't get a pair request even on Android 9.  I don't think he will get a pair request on his Android 6 phone.

**Reply by kresimirjedud** [reply]
@jamorham I found this mobile phone on the list of tested phones for AAPS  and Dexcome 6 (it should work with DExcome One as well), therefore I find this discussion closed. Tomorrow I'm getting the phone. Hopefully I will be able to install and setup everything by myself. I cannot thank you enough for your time, you helped me a lot!
![Capture](https://user-images.githubusercontent.com/71585534/217660519-446ffbf9-c543-4193-8100-a1097a43da8a.PNG)

**Reply by Navid200** [reply]
I would not buy a phone before confirming, using the Dexcom receiver, that there is nothing wrong with the transmitter.

Even if you buy a new phone, there is no guarantee Dexcom One C will work with it.  This is a new feature.

**Reply by kresimirjedud** [reply]
@Navid200 thank you for your concern. I have just paired a transmitter and a receiver. The sensor is warming up. I assume it will end up OK. I will let you know.

**Reply by kresimirjedud** [reply]
@Navid200 the sensor was started successfully with the Dexcom receiver and it functions well. Today I will get a phone and try to connect it to xDrip.

**Reply by kresimirjedud** [reply]
**@Navid200 @jamorham** I just want to confirm that I managed to set up everything on **Cubot Kingkong Mini2 pro** mobile without any problems. It seems as if it works like a charm ;) I used the latest xdrip (08.02.2023) and I can confirm **Dexcome One C transmitter connected almost automatically**, without any problems. I was really surprised. And **@jamorham** thank you for recommending this phone, it is exactly what I wanted. I have never heard of it but this is a small and rather cheap phone, with Android 11 and only 4" screen. **Perfect for a child!** Once again guys I am really grateful for your help. Also, the more I look at xDrip the more I see it is the state of the art piece of software. Incredible! Now I can finally send the little one to school!

**Reply by Navid200** [reply]
Thanks for the feedback.

It's great that you were able to solve the problem.  It is fantastic.

**Reply by jamorham** [reply]
@kresimirjedud great to hear you are getting good results. Please update us again in the future because the reason I was looking at this phone is because of a report that it worked for a week and then suddenly stopped being able to connect to the transmitter. Please also confirm the exact software release on the phone. Simply swipe down from the notification area and it will say like 10 (QP1A...) please let me know the numbers there so we know which version it is. The one I'm testing I haven't updated to the latest release yet but also has been flawless. (also there may be a difference between Mini2 and Mini2 pro)

**Reply by kresimirjedud** [reply]
@jamorham attached please find the phone status page. It is:
- Android 11
- Android security update 05. July 2022
- Google Play security update 01. January 2023
- Base version MOLY-LR 12A .......(see the photo)
- Core version 4.19.127 .......(see the photo)
- Intermediate version CUBOT_KINGKONG MINI2 .......(see the photo)

So far it all functions well. I think it performs better than MiaoMiao and Libre1+xDrip since we had constant connection issues with it. In this constellation (DexcomeOne+xDrip) when the phone is out of range it connects by itself. If I understood it well DexOne functions the way it connects, sends the data and disconnects (most probably that is why the battery can last for 3 months), and MiaoMiao was connected all the time ...

What I also like is that I have found this phone on the list of tested AAPS phones (https://docs.google.com/spreadsheets/d/1zO-Vf3wv0jji5Gflk6pe48oi348ApF5RvMcI6NG5TnY/edit#gid=2097219952) and we plan to implement AAPS as well. 

I will put a reminder to send you feedback within 10 days. Should you need any other info please do not hesitate to contact me. 

![king_kong_mini_pro_version](https://user-images.githubusercontent.com/71585534/218265906-f9339a1b-24d7-4d2d-a254-c11ec0559949.jpeg)

**Reply by kresimirjedud** [reply]
@jamorham as promised I'm sending you feedback for King Kong Mini Pro and Dexcome One (transmitter starting with C ). We are using the second sensor now. Everything seems to function excellently. The connectivity is no more an issue, as it was before when we used Libre1 and MiaoMiao. Perhaps the only issue I noticed is that when the phone is out of range for more than 4 readings (20+ minutes) then it does not connect automatically back. In that case I need to trigger the connectivity by changing the transmitter id and then bringing it back. This is actually a bit problematic due to a fact that the phone is owned by a child. With Libre and MiaoMiao it was easy for him to turn off BLuetooth and go under Bluetooth and just click the blue coloured device. This is something very difficult for me to explain him, and quite risky, too. My question is: would it be possible to have just one place (e.g. one button) somewhere in the menu which could: change the transmitter id to something and then change it back to as it was. For a child this would be perfect to handle if this situation occurs. Apart from this everything else is functioning. Thank you for your effort and advice!
